deepinv/tests/test_sampling.py

```diff
@@ -17,84 +16,109 @@ def device():
 
 @pytest.fixture
 def imsize():
-    h = 28
-    w = 32
-    c = 3
+    h = 2
+    w = 2
+    c = 1
     return c, h, w
 
 
-# TODO: use a DummyCircle as dataset and check convergence of optim algorithms (maybe with TV denoiser)
-@pytest.fixture
-def dummy_dataset(imsize, device):
-    return DummyCircles(samples=1, imsize=imsize)
-
-
-def choose_algo(algo, prior, likelihood, stepsize, thresh_conv, sigma):
+def choose_algo(algo, likelihood, thresh_conv, sigma, sigma_prior):
     if algo == "ULA":
         out = ULA(
-            prior,
+            GaussianScore(sigma_prior),
             likelihood,
-            max_iter=1000,
+            max_iter=100,
+            thinning=1,
             verbose=True,
-            alpha=0.9,
-            step_size=0.01 * stepsize,
-            clip=(-1, 2),
+            step_size=0.01 / (1 / sigma**2 + 1 / sigma_prior**2),
+            clip=(-100, 100),
             thresh_conv=thresh_conv,
-            sigma=sigma,
+            sigma=1,
         )
     elif algo == "SKRock":
         out = SKRock(
-            prior,
+            GaussianScore(sigma_prior),
             likelihood,
-            max_iter=100,
+            max_iter=200,
             verbose=True,
             thresh_conv=thresh_conv,
-            alpha=0.9,
-            step_size=stepsize,
-            clip=(-1, 2),
-            sigma=sigma,
+            inner_iter=5,
+            step_size=1 / (1 / sigma**2 + 1 / sigma_prior**2),
+            clip=(-100, 100),
+            sigma=1,
+        )
+    elif algo == "DDRM":
+        diff = dinv.sampling.DDRM(
+            denoiser=GaussianDenoiser(sigma_prior),
+            sigma_noise=sigma,
+            eta=1,
+            sigmas=np.linspace(1, 0, 100),
         )
+        out = dinv.sampling.DiffusionSampler(diff, clip=(-100, 100), max_iter=100)
```

Comment:
This test failed here in the last push with no apparent reason; maybe we can set a random seed to avoid having some random failuers?

Reply:
Yes, good idea